Update index.md (#514)
aws/aws-eks-best-practices: content/upgrades/index.md

Managed node groups automate the provisioning and lifecycle management of nodes. This means that you can create, automatically update, or terminate nodes with a single operation.

> For Karpenter data plane updgrades, refer to below sections:

Reviewer: ```suggestion
For Karpenter data plane upgrades, refer to below sections:
```

> * [Use Drift feature for Karpenter managed nodes](#use-drift-feature-for-karpenter-managed-nodes)

Reviewer: consider just calling it drift
```suggestion
* [Use Drift for Karpenter managed nodes](#use-drift-for-karpenter-managed-nodes)
```

> * [Use Node expiry for Karpenter managed nodes](#use-node-expiry-for-karpenter-managed-nodes)

Reviewer: consider using the same name as the configuration field
```suggestion
* [Use ExpireAfter for Karpenter managed nodes](#use-expire-after-for-karpenter-managed-nodes)
```

* **Kubernetes < v1.28** — If you are using a version older than v1.28, the supported skew between the API server and the kubelet is n-2. For example, if your EKS version is 1.27, the oldest kubelet version you can use is 1.25. This version skew is applicable across [AWS Fargate](https://docs.aws.amazon.com/eks/latest/userguide/fargate.html), [managed node groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html), and [self-managed nodes](https://docs.aws.amazon.com/eks/latest/userguide/worker.html).

> ## Use Drift feature for Karpenter managed nodes

Reviewer: will need to update this title if you accept the above suggestion with the different anchorlink
```suggestion
## Use Drift for Karpenter managed nodes
```

> Karpenter’s [Drift feature](https://karpenter.sh/docs/concepts/disruption/#drift) can automatically upgrade the Karpenter-provisioned nodes to stay in-sync with the EKS control plane. Refer to [How to upgrade an EKS Cluster with Karpenter](https://karpenter.sh/docs/faq/#how-do-i-upgrade-an-eks-cluster-with-karpenter) for more details.

Reviewer: ```suggestion
Karpenter’s [Drift](https://karpenter.sh/docs/concepts/disruption/#drift) can automatically upgrade the Karpenter-provisioned nodes to stay in-sync with the EKS control plane. Refer to [How to upgrade an EKS Cluster with Karpenter](https://karpenter.sh/docs/faq/#how-do-i-upgrade-an-eks-cluster-with-karpenter) for more details.
```

I realized we also need to update the link that you've linked here. Let me fix that before we merge this.

This means that if the AMI ID specified in the Karpenter EC2 Nodeclass is updated, Karpenter will detect the drift and start replacing the nodes with the new AMI. 
To understand how Karpenter manages AMIs and the different options available to Karpenter users to control the AMI upgrade process see the documentation on [how to manage AMIs in Karpenter](https://karpenter.sh/docs/tasks/managing-amis/).

> [Karpenter can be configured to use custom AMIs.](https://karpenter.sh/docs/concepts/nodeclasses/) If you use custom AMIs with Karpenter, you are responsible for the version of kubelet. 

Reviewer: I think you can probably remove this line, since it's encompassed in the managing-amis link

Author: ok will remove, I did not see the mention of custom AMI, hence added.

> ## Use Node expiry for Karpenter managed nodes

Reviewer: ```suggestion
## Use ExpireAfter for Karpenter managed nodes
```

> Karpenter will mark nodes as expired and disrupt them after they have lived a set number of seconds, based on the NodePool’s `spec.disruption.expireAfter` value. This node expiry helps to reduce security vulnerabilities and issues that can arise from long-running nodes, such as file fragmentation or memory leaks. When you set a value for expireAfter in your NodePool, this activates node expiry. For more information, see [Disruption](https://karpenter.sh/docs/concepts/disruption/#methods) on the Karpenter website.

Reviewer: expireAfter is actually of type time.Duration, which allows compound durations like `10m5s`. 
```suggestion
Karpenter will mark nodes as expired and disrupt them after they have lived the duration specified in `spec.disruption.expireAfter. This node expiry helps to reduce security vulnerabilities and issues that can arise from long-running nodes, such as file fragmentation or memory leaks. When you set a value for expireAfter in your NodePool, this activates node expiry. For more information, see [Disruption](https://karpenter.sh/docs/concepts/disruption/#methods) on the Karpenter website.
```

> One way Karpenter implements node upgrades is using the concept of node expiry. This reduces the planning required for node upgrades. Karpenter will mark nodes as expired and disrupt them after they have lived a set number of seconds, based on the NodePool’s `spec.disruption.expireAfter` value. This node expiry helps to reduce security vulnerabilities and issues that can arise from long-running nodes, such as file fragmentation or memory leaks. When you set a value for expireAfter in your NodePool, this activates node expiry. For more information, see [Disruption](https://karpenter.sh/docs/concepts/disruption/#methods) on the Karpenter website.

Reviewer: The more common use-case for ExpireAfter is to recycle nodes due to a security requirement on node uptime. Let's order it so that it's mentioned that way first, and offhand included as an upgrade path after. 
```suggestion
Karpenter will mark nodes as expired and disrupt them after they have lived a set number of seconds, based on the NodePool’s `spec.disruption.expireAfter` value. This node expiry helps to reduce security vulnerabilities and issues that can arise from long-running nodes, such as file fragmentation or memory leaks. When you set a value for expireAfter in your NodePool, this activates node expiry. For more information, see [Disruption](https://karpenter.sh/docs/concepts/disruption/#methods) on the Karpenter website.

If you're using automatic AMI upgrades, ExpireAfter can periodically refresh and upgrade your nodes.
```
